app/src/main/java/com/github/se/icebreakrr/ui/sections/Notifications.kt

```diff
@@ -99,57 +104,88 @@ fun NotificationScreen(
-                modifier =
-                    Modifier.padding()
-                        .padding(horizontal = HORIZONTAL_PADDING)
-                        .testTag("notificationScroll")) {
-                  item {
-                    Text(
-                        text = MEETING_REQUEST_SENT,
-                        fontWeight = FontWeight.Bold,
-                        modifier =
-                            Modifier.padding(vertical = TEXT_VERTICAL_PADDING)
-                                .testTag("notificationFirstText"))
-                    Column(verticalArrangement = Arrangement.spacedBy(CARD_SPACING)) {
-                      sentCardList.value.forEach { p ->
-                        ProfileCard(profile = p, onclick = {}, greyedOut = true)
+            }
+            MeetingRequestOption.SENT -> {
+              LazyColumn(
+                  modifier =
+                      Modifier.padding()
+                          .padding(horizontal = HORIZONTAL_PADDING)
+                          .testTag("notificationScroll")) {
+                    item {
+                      Text(
+                          text = MEETING_REQUEST_SENT,
+                          fontWeight = FontWeight.Bold,
+                          modifier =
+                              Modifier.padding(vertical = TEXT_VERTICAL_PADDING)
+                                  .testTag("notificationFirstText"))
+                      Column(verticalArrangement = Arrangement.spacedBy(CARD_SPACING)) {
+                        sentCardList.value.forEach { p ->
+                          ProfileCard(profile = p, onclick = {}, greyedOut = true)
+                        }
                       }
                     }
                   }
-                }
+            }
+            MeetingRequestOption.CHOOSE_LOCATION -> {
+              LazyColumn(
+                  modifier =
+                      Modifier.padding()
+                          .padding(horizontal = HORIZONTAL_PADDING)
+                          .testTag("notificationScroll")) {
+                    item {
+                      Text(
+                          text = MEETING_REQUEST_LOCATION_PENDING,
+                          fontWeight = FontWeight.Bold,
+                          modifier =
+                              Modifier.padding(vertical = TEXT_VERTICAL_PADDING)
+                                  .testTag("notificationFirstText"))
+                      Column(verticalArrangement = Arrangement.spacedBy(CARD_SPACING)) {
+                        pendingLocation.value.forEach { p ->
+                          ProfileCard(
+                              profile = p,
+                              onclick = {
+                                navigationActions.navigateTo(
+                                    Screen.MAP_MEETING_LOCATION_SCREEN + "?userId=${p.uid}")
+                              },
+                              greyedOut = false)
+                        }
```

Comment:
The LazyColumn blocks for INBOX, SENT, and CHOOSE_LOCATION options share similar code so you could extract the common logic into a reusable composable function.

Reply:
That's a good refactoring, thank you. It has been done